### ISSUE TITLE
Node runners should be able to schedule a halt height

### ISSUE THREAD

**Issue by cmwaters** [tendermint/tendermint#5645]
## Summary

This would assist with scheduling upgrades. Generally I feel that we should reconsider bringing some of the upgrading utilities written in the cosmos-sdk upstream to tendermint where a broader range of users can benefit from them. 

## Proposal

The following was proposed in [ADR 016 ](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-016-protocol-versions.md)

It may be valuable to support an `/unsafe_stop?height=_` endpoint to tell Tendermint to shutdown at a given height. This could be used by an external manager process that oversees upgrades by checking out and installing new software versions and restarting the process. It would subscribe to the relevant upgrade event (needs to be implemented) and call `/unsafe_stop` at the correct height (of course only after getting approval from its user!)

There may be better ways of implementing this depending on how the overall path for upgrading pans out

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by alexanderbez** [comment]
ACK. Note, we should ensure this feature works in harmony with the SDK's `halt-time` and `halt-height` configurations, which do a deferred SIGINT after `ABCI#Commit`.
